library/RemoteServer.cpp

```diff
@@ -476,17 +478,33 @@ void ServerMainImpl::threadFn(std::promise<bool> promise, int port)
     CActiveSocket *client = nullptr;
 
     try {
-        while ((client = server.socket.Accept()) != NULL)
+        for (int acceptFail = 0 ; server.socket.IsSocketValid() && acceptFail < 5 ; acceptFail++)
         {
-            BlockGuard lock;
-            ServerConnection::Accepted(client);
-            client = nullptr;
+            if ((client = server.socket.Accept()) != NULL)
+            {
+                BlockGuard lock;
+                ServerConnection::Accepted(client);
+                client = nullptr;
+            }
+            else
+            {
+                WARN(socket).print("Accepting connection error: %s %d of %d\n", server.socket.DescribeError(), acceptFail + 1, 5);
+            }
         }
     } catch(BlockedException &) {
         if (client)
             client->Close();
         delete client;
     }
+
+    if (server.socket.IsSocketValid())
+    {
+        WARN(socket).print("To many failed accept, shutting down RemoteServer \n");
```

Comment:
s/To/Too/

Reply:
I would also suggest `s/accept/accepts/`

```suggestion
        WARN(socket).print("Too many failed accepts, shutting down RemoteServer\n");
```